examples/keras_io/vision/image_classifier.py

```diff
@@ -0,0 +1,692 @@
+`EfficientNetV2B0(weights="imagenet")`. While the old API was great for
+classification, it did not scale effectively to other use cases that required
+complex architectures, like object deteciton and semantic segmentation.
+
+Now that our classifier is built, let's apply it to this cute cat picture!
+"""
+
+filepath = keras.utils.get_file(origin="https://i.imgur.com/9i63gLN.jpg")
+image = keras.utils.image_utils.load_img(filepath)
+image = np.array(image)
+keras_cv.visualization.plot_image_gallery(
+    [image], rows=1, cols=1, value_range=(0, 255), show=True, scale=4
+)
+
+"""Next, let's get some predictions from our classifier:"""
+
+predictions = classifier.predict(np.expand_dims(image, axis=0))
+
+"""Predictions come in the form of softmax-ed category rankings.
+We can find the index of the top classes using a simple argsort function:
+"""
+
+top_classes = predictions[0].argsort(axis=-1)
+
+"""In order to decode the class mappings, we can construct a mapping from
+category indices to ImageNet class names.
+For convenience, I've stored the ImageNet class mapping in a GitHub gist.
+Let's download and load it now.
+"""
+
+classes = keras.utils.get_file(
+    origin="https://gist.githubusercontent.com/LukeWood/62eebcd5c5c4a4d0e0b7845780f76d55/raw/fde63e5e4c09e2fa0a3436680f436bdcb8325aac/ImagenetClassnames.json"
+)
+with open(classes, "rb") as f:
+    classes = json.load(f)
+
+"""Now we can simply look up the class names via index:"""
+
+top_two = [classes[str(i)] for i in top_classes[-2:]]
+print("Top two classes are:", top_two)
+
+"""Great!  Both of these appear to be correct!
+However, one of the classes is "Velvet".
+We're trying to classify Cats VS Dogs.
+We don't care about the velvet blanket!
+
+Ideally, we'd have a classifier that only performs computation to determine if
+an image is a cat or a dog, and has all of its resources dedicated to this
+task. This can be solved by fine tuning our own classifier.
+
+# Fine tuning a pretrained classifier
+
+When labeled images specific to our task are available, fine-tuning a custom
+classifier can improve performance.
+If we want to train a Cats vs Dogs Classifier, using explicitly labeled Cat vs
+Dog data should perform better than the generic classifier!
+For many tasks, no relevant pretrained model
+will be available (e.g., categorizing images specific to your application).
+
+First, let's get started by loading some data:
+"""
+
+BATCH_SIZE = 32
+IMAGE_SIZE = (224, 224)
+AUTOTUNE = tf.data.AUTOTUNE
+tfds.disable_progress_bar()
+
+data, dataset_info = tfds.load(
+    "cats_vs_dogs",
+    with_info=True,
+    as_supervised=True
+)
+train_steps_per_epoch = (
+    dataset_info.splits["train"].num_examples // BATCH_SIZE
+)
+train_dataset = data["train"]
+
+num_classes = dataset_info.features["label"].num_classes
+
+resizing = keras_cv.layers.Resizing(
+    IMAGE_SIZE[0], IMAGE_SIZE[1], crop_to_aspect_ratio=True
+)
+
+
+def preprocess_inputs(image, label):
+    image = tf.cast(image, tf.float32)
```

Comment:
Since the example has quite a bit of TF-only APIs...

1. We could reduce or remove the TF dependency and keep it in the backend-agnostic folder.
2. We could move it to the TF-only folder.

Reply:
The TF components are just for preprocessing which for now has to live in tf.data. I'm fine with moving it to TF-only, but I would expect this to work fine with all backends since this is constrained to the tf.data parts of the example

Reply:
I was trying your suggestions out @fchollet and seems like we would still need the `tf` import for one hot encoding since `CategoryEncoding` doesn't accept `float32` labels (so, even if we use it, we'd still need to cast labels to integers).

Other than that, everything else works. Let me know if this is OK from your side!